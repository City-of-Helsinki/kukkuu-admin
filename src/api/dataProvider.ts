import { getVenues } from '../domain/venues/api/VenueApi';
import {
  MethodHandlers,
  Method,
  Resource,
  DataProviderParams as Params,
} from './types';

const METHOD_HANDLERS: MethodHandlers = {
  venues: {
    LIST: getVenues,
  },
};

const runHandler = (method: Method, resource: Resource, params: Params) => {
  const handlers = METHOD_HANDLERS[resource];
  if (handlers === undefined) {
    throw new Error(`Invalid resource "${resource}".`);
  }

  const handler = handlers[method];
  if (handler === undefined) {
    throw new Error(
      `Method "${method}" for resource "${resource}" is not implemented.`
    );
  }

  return handler(params);
};

const dataProvider = {
  getList: async (resource: Resource, params: Params) => {
    const data = await runHandler('LIST', resource, params);
    return {
      data,
      total: data.length,
    };
  },
};

export default dataProvider;
