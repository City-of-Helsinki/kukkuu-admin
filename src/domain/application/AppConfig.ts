class AppConfig {
  static get apiUrl() {
    return getEnvOrError(process.env.REACT_APP_API_URI, 'REACT_APP_API_URI');
  }

  static get oidcAuthority() {
    const origin = getEnvOrError(
      process.env.REACT_APP_OIDC_AUTHORITY,
      'REACT_APP_OIDC_AUTHORITY'
    );
    return new URL(origin).href;
  }

  /**
   * The audiences used in the OIDC.
   * 
   * @example
   * // In Tunnistamo it can be left undefined.
   * ["https://api.hel.fi/auth/kukkuu"]
   * // In Keycloak:
   * [
        'kukkuu-api-test',
        'profile-api-test',
      ]
   */
  static get oidcAudience() {
    return process.env.REACT_APP_OIDC_AUDIENCES;
  }

  static get oidcClientId() {
    return getEnvOrError(
      process.env.REACT_APP_OIDC_CLIENT_ID,
      'REACT_APP_OIDC_CLIENT_ID'
    );
  }

  static get oidcScope() {
    return getEnvOrError(
      process.env.REACT_APP_OIDC_SCOPE,
      'REACT_APP_OIDC_SCOPE,'
    );
  }

  static get oidcReturnType() {
    // "code" for authorization code flow.
    return process.env.REACT_APP_OIDC_RETURN_TYPE ?? 'code';
  }

  static get oidcKukkuuApiClientId() {
    return (
      process.env.REACT_APP_OIDC_KUKKUU_API_CLIENT_ID ?? this.oidcKukkuuAPIScope
    );
  }

  static get oidcKukkuuApiTokensUrl() {
    return this.oidcServerType === 'KEYCLOAK'
      ? `${this.oidcAuthority}protocol/openid-connect/token`
      : `${this.oidcAuthority}api-tokens/`;
  }

  static get oidcKukkuuAPIScope() {
    return getEnvOrError(
      process.env.REACT_APP_KUKKUU_API_OIDC_SCOPE,
      'REACT_APP_KUKKUU_API_OIDC_SCOPE'
    );
  }

  /**
   * NOTE: The oidcServerType is not an OIDC client attribute.
   * It's purely used to help to select a configuration for the LoginProvider.
   * */
  static get oidcServerType(): 'KEYCLOAK' | 'TUNNISTAMO' {
    const oidcServerType =
      process.env.REACT_APP_OIDC_SERVER_TYPE ?? 'TUNNISTAMO';
    if (!['KEYCLOAK', 'TUNNISTAMO'].includes(oidcServerType)) {
      throw new Error(`Invalid OIDC server type: ${oidcServerType}`);
    }
    return oidcServerType as 'KEYCLOAK' | 'TUNNISTAMO';
  }
}

// Accept both variable and name so that variable can be correctly replaced
// by build.
// process.env.VAR => value
// process.env["VAR"] => no value
// Name is used to make debugging easier.
function getEnvOrError(variable?: string, name?: string) {
  if (!variable) {
    throw Error(`Environment variable with name ${name} was not found`);
  }
  return variable;
}

export default AppConfig;
