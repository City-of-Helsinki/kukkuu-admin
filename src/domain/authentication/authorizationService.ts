import { MyAdminProfile_myAdminProfile_projects_edges_node as ProjectNode } from '../../api/generatedTypes/MyAdminProfile';
import RelayList from '../../api/relayList';
import dataProvider from '../../api/dataProvider';
import projectService from '../profile/profileService';

export const PERMISSIONS = 'permissions';

const ProjectList = RelayList<ProjectNode>();

export class AuthorizationService {
  constructor() {
    this.fetchRole = this.fetchRole.bind(this);
    this.isAuthorized = this.isAuthorized.bind(this);
    this.getRole = this.getRole.bind(this);
    this.clear = this.clear.bind(this);
  }

  async fetchRole(): Promise<void> {
    try {
      const { data } = await dataProvider.getMyAdminProfile();
      const projects = ProjectList((data as any)?.projects).items;

      projectService.setDefaultProjectId(projects);
      sessionStorage.setItem(PERMISSIONS, 'admin');
    } catch (e) {
      sessionStorage.setItem(PERMISSIONS, 'none');
    }
  }

  isAuthorized(): boolean {
    return sessionStorage.getItem(PERMISSIONS) !== null;
  }

  getRole(): string | null {
    return sessionStorage.getItem(PERMISSIONS);
  }

  clear() {
    sessionStorage.removeItem(PERMISSIONS);
  }
}

export default new AuthorizationService();
