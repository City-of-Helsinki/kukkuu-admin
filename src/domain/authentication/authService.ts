import {
  type User,
  type UserManagerSettings,
  UserManager,
  Log,
  WebStorageStateStore,
} from 'oidc-client-ts';
import axios from 'axios';
import * as Sentry from '@sentry/browser';

import projectService from '../projects/projectService';
import authorizationService from './authorizationService';
import AppConfig from '../application/AppConfig';

const origin = window.location.origin;
export const API_TOKEN = 'apiToken';
export const REFRESH_TOKEN = 'refreshToken';

export type ApiTokenClientProps = {
  url: string;
  queryProps?: { permission: string; grant_type: string };
  audiences?: string[];
};

export class AuthService {
  private readonly userManager: UserManager;
  private readonly apiTokensClientConfig: ApiTokenClientProps;
  private readonly authServerType: 'KEYCLOAK' | 'TUNNISTAMO';
  private readonly audience: string;

  constructor() {
    this.authServerType = AppConfig.oidcServerType;
    this.audience = AppConfig.oidcAudience ?? AppConfig.oidcKukkuuApiClientId;

    const settings: UserManagerSettings = {
      loadUserInfo: true,
      userStore: new WebStorageStateStore({ store: window.localStorage }),
      response_type: AppConfig.oidcReturnType,
      authority: AppConfig.oidcAuthority,
      client_id: AppConfig.oidcClientId,
      scope: AppConfig.oidcScope,
      redirect_uri: `${origin}/callback`,
      post_logout_redirect_uri: `${origin}/`,
      automaticSilentRenew: AppConfig.oidcAutomaticSilentRenew,
      checkSessionIntervalInSeconds: AppConfig.oidcSessionPollerIntervalInMs,
      silent_redirect_uri: `${origin}/silent_renew.html`,
      revokeTokensOnSignout: true,
    };

    if (!settings.automaticSilentRenew) {
      // eslint-disable-next-line no-console
      console.info('Auth token silent renew is disabled.');
    }

    if (process.env.NODE_ENV === 'development') {
      // Show oidc debugging info in the console only while developing
      Log.setLogger(console);
      Log.setLevel(Log.INFO);
    }

    // User Manager instance
    this.userManager = new UserManager(settings);

    // Api tokens client configuration
    this.apiTokensClientConfig = {
      url: AppConfig.oidcKukkuuApiTokensUrl,
      queryProps:
        this.authServerType === 'KEYCLOAK'
          ? {
              permission: '#access',
              grant_type: 'urn:ietf:params:oauth:grant-type:uma-ticket',
            }
          : undefined,
      audiences:
        this.authServerType === 'KEYCLOAK' && AppConfig.oidcAudience
          ? [AppConfig.oidcAudience]
          : undefined,
    };

    // Public methods
    this.getUser = this.getUser.bind(this);
    this.getToken = this.getToken.bind(this);
    this.getRefreshToken = this.getRefreshToken.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);
    this.login = this.login.bind(this);
    this.endLogin = this.endLogin.bind(this);
    this.renewToken = this.renewToken.bind(this);
    this.logout = this.logout.bind(this);
    this.resetAuthState = this.resetAuthState.bind(this);
    this.fetchApiToken = this.fetchApiToken.bind(this);

    // Events
    this.userManager.events.addAccessTokenExpired(() => {
      this.logout();
    });

    this.userManager.events.addSilentRenewError(() => {
      this.logout();
    });

    this.userManager.events.addUserSignedOut(() => {
      this.resetAuthState();
    });

    this.userManager.events.addUserLoaded(async (user) => {
      await this.fetchApiToken(user);
      await authorizationService.fetchRole();
    });
  }

  public async getUser(): Promise<User | null> {
    return await this.userManager.getUser();
  }

  public getToken(): string | null {
    return localStorage.getItem(API_TOKEN);
  }

  public getRefreshToken(): string | null {
    return localStorage.getItem(REFRESH_TOKEN);
  }

  public getUserStorageKey(): string {
    return `oidc.user:${AppConfig.oidcAuthority}:${AppConfig.oidcClientId}`;
  }

  public isAuthenticated() {
    const userKey = this.getUserStorageKey();
    const oidcStorage = localStorage.getItem(userKey);
    const apiTokens = this.getToken();

    return (
      !!oidcStorage && !!JSON.parse(oidcStorage).access_token && !!apiTokens
    );
  }

  public async login(path = '/'): Promise<void> {
    try {
      await this.userManager.signinRedirect({ url_state: path });
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error);
      if (error instanceof Error) {
        if (error.message !== 'Network Error') {
          Sentry.captureException(error);
        }
      }
    }
  }

  public async endLogin(): Promise<User> {
    // eslint-disable-next-line no-console
    console.info('Ending the login...');
    const user = await this.userManager.signinRedirectCallback();

    await this.fetchApiToken(user);
    await authorizationService.fetchRole();

    return user;
  }

  public async renewToken(): Promise<User | null> {
    // eslint-disable-next-line no-console
    console.info('Renewing the JWT token...');
    const user = await this.userManager.signinSilent();
    if (user) {
      localStorage.setItem(API_TOKEN, user.access_token);
      if (user.refresh_token) {
        localStorage.setItem(REFRESH_TOKEN, user.refresh_token);
      } else {
        localStorage.removeItem(REFRESH_TOKEN);
      }
    }
    return user;
  }

  public resetAuthState() {
    // eslint-disable-next-line no-console
    console.warn('Resetting the auth state...');
    localStorage.removeItem(API_TOKEN);
    localStorage.removeItem(REFRESH_TOKEN);
    projectService.clear();
    this.userManager.clearStaleState();
    authorizationService.clear();
  }

  public async logout(): Promise<void> {
    // eslint-disable-next-line no-console
    console.info('Logout...');
    this.resetAuthState();
    await this.userManager.signoutRedirect();
  }

  /**
   * Query the API tokens endpoint with the given access token.
   * @param accessToken The access token to use for the API tokens endpoint query.
   * @returns For Tunnistamo should return a dictionary with the API identifiers
   * as the keys and the API tokens as the values here as data (See [1]).
   * For Keycloak should return access_token, token_type, and optionally
   * expires_in, refresh_token and scope here as data (See [2, 3]).
   *
   * [1] Tunnistamo "Obtaining the API tokens":
   * https://github.com/City-of-Helsinki/tunnistamo/blob/r211109/tokens.rst#obtaining-the-api-tokens
   *
   * [2] OIDC 1.0 "Authentication > Token Endpoint > Successful Token Response":
   * https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse
   *
   * [3] OAuth 2.0 "Issuing an Access Token > Successful Response":
   * https://www.rfc-editor.org/rfc/rfc6749.html#section-5.1
   */
  private async queryApiTokensEndpoint(accessToken: string) {
    return await axios(this.apiTokensClientConfig.url, {
      method: 'post',
      baseURL: AppConfig.oidcAuthority,
      headers: {
        Authorization: `bearer ${accessToken}`,
        Accept: 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
      },
      data:
        this.authServerType === 'KEYCLOAK'
          ? {
              audience: this.audience,
              ...this.apiTokensClientConfig.queryProps,
            }
          : {},
    });
  }

  private async fetchApiToken(user: User): Promise<void> {
    const accessToken = user.access_token;
    try {
      const { data } = await this.queryApiTokensEndpoint(accessToken);
      const apiToken =
        this.authServerType === 'KEYCLOAK'
          ? data.access_token
          : data[AppConfig.oidcKukkuuApiClientId];

      // NOTE: Currently only supporting refresh tokens with Keycloak.
      // Tunnistamo does not return a refresh token from the API tokens
      // endpoint, but Keycloak does from the token endpoint.
      const refreshToken =
        (this.authServerType === 'KEYCLOAK' ? data.refresh_token : null) ??
        user.refresh_token;

      localStorage.setItem(API_TOKEN, apiToken);
      if (!refreshToken) {
        localStorage.removeItem(REFRESH_TOKEN);
      } else {
        localStorage.setItem(REFRESH_TOKEN, refreshToken);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error('Failed to fetch API token', error);
      Sentry.captureException(error);
    }
  }
}

const authServiceInstance = new AuthService();

export default authServiceInstance;
