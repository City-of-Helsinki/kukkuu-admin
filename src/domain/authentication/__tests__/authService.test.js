import axios from 'axios';

import authService, { API_TOKEN, REFRESH_TOKEN } from '../authService';
import authorizationService from '../authorizationService';
import AppConfig from '../../application/AppConfig';
import dataProvider from '../../../api/dataProvider';

jest.mock('axios');

const testProjectId = btoa('ProjectNode:234');

const getOidcUserKey = () =>
  `oidc.user:${AppConfig.oidcAuthority}:${AppConfig.oidcClientId}`;

describe('authService', () => {
  const userManager = authService.userManager;

  beforeEach(() => {
    jest.spyOn(dataProvider, 'getMyAdminProfile').mockResolvedValue({
      data: {
        id: btoa('AdminNode:123'),
        projects: {
          edges: [
            {
              node: {
                id: testProjectId,
                year: 2023,
                name: 'test project 2023',
                myPermissions: {
                  publish: true,
                  manageEventGroups: true,
                },
              },
            },
          ],
        },
      },
    });
  });

  afterEach(() => {
    localStorage.clear();
    localStorage.setItem.mockClear();
    jest.restoreAllMocks();
  });

  describe('getUser', () => {
    it('should resolve to the user value which has been resolved from getUser', async () => {
      expect.assertions(1);
      const mockUser = { name: 'Sam Littel' };

      jest.spyOn(userManager, 'getUser').mockResolvedValueOnce(mockUser);

      const user = await authService.getUser();

      expect(user).toBe(mockUser);
    });
  });

  describe('getToken', () => {
    it('should get API_TOKEN from localStorage', () => {
      const origCallCount = localStorage.getItem.mock.calls.length;
      authService.getToken();

      expect(localStorage.getItem).toHaveBeenLastCalledWith(API_TOKEN);
      expect(localStorage.getItem).toHaveBeenCalledTimes(origCallCount + 1);
    });
  });

  describe('getRefreshToken', () => {
    it('should get REFRESH_TOKEN from localStorage', () => {
      const origCallCount = localStorage.getItem.mock.calls.length;
      authService.getRefreshToken();

      expect(localStorage.getItem).toHaveBeenLastCalledWith(REFRESH_TOKEN);
      expect(localStorage.getItem).toHaveBeenCalledTimes(origCallCount + 1);
    });
  });

  describe('isAuthenticated', () => {
    it('should return false if no token can be found', () => {
      jest.spyOn(authService, 'getToken').mockReturnValue(null);

      expect(authService.isAuthenticated()).toBe(false);
    });

    it("should return false if oidc user from sessionStorage doesn't exist", () => {
      const apiTokens = '5ed3abc5-9b65-4879-8d09-3cd8499650ef';
      jest.spyOn(authService, 'getToken').mockReturnValue(apiTokens);
      sessionStorage.clear();

      expect(authService.isAuthenticated()).toBe(false);
    });

    it("should return false if oidc user from sessionStorage doesn't have an access_token property", () => {
      const apiTokens = '5ed3abc5-9b65-4879-8d09-3cd8499650ef';
      const invalidUser = JSON.stringify({});

      jest.spyOn(authService, 'getToken').mockReturnValue(apiTokens);
      sessionStorage.setItem(getOidcUserKey(), invalidUser);

      expect(authService.isAuthenticated()).toBe(false);
    });

    it('should return true if oidc user is valid and tokens are returned from getToken', () => {
      const apiToken = '5ed3abc5-9b65-4879-8d09-3cd8499650ef';
      const validUser = JSON.stringify({
        name: 'Mr. Louisa Tromp',
        access_token: '5ed3abc5-9b65-4879-8d09-3cd8499650ef',
      });

      jest.spyOn(authService, 'getToken').mockReturnValue(apiToken);
      localStorage.setItem(getOidcUserKey(), validUser);

      expect(authService.isAuthenticated()).toBe(true);
    });
  });

  describe('login', () => {
    it('should call signinRedirect from oidc with the provided path', () => {
      const path = '/applications';
      const signinRedirect = jest.spyOn(userManager, 'signinRedirect');

      authService.login(path);

      expect(signinRedirect).toHaveBeenNthCalledWith(1, {
        url_state: path,
      });
    });
  });

  describe('endLogin', () => {
    axios.mockResolvedValue({ data: {} });
    const access_token = 'db237bc3-e197-43de-8c86-3feea4c5f886';
    const refresh_token = 'ec3510ee-11be-46d1-8b6a-ab97cb29b169';
    const mockUser = {
      name: 'Penelope Krajcik',
      access_token,
      refresh_token,
    };

    it('should call signinRedirectCallback from oidc', () => {
      const signinRedirectCallback = jest
        .spyOn(userManager, 'signinRedirectCallback')
        .mockImplementation(() => Promise.resolve(mockUser));

      authService.endLogin();

      expect(signinRedirectCallback).toHaveBeenCalledTimes(1);
    });

    it('should call fetchRoles from authorizationService', async () => {
      expect.assertions(1);
      jest
        .spyOn(userManager, 'signinRedirectCallback')
        .mockReturnValue(Promise.resolve(mockUser));

      const fetchRoleSpy = jest
        .spyOn(authorizationService, 'fetchRole')
        .mockImplementation(() => Promise.resolve());

      await authService.endLogin();

      expect(fetchRoleSpy).toHaveBeenCalledTimes(1);
    });

    it('should return the same user object returned from signinRedirectCallback', async () => {
      expect.assertions(1);
      jest
        .spyOn(userManager, 'signinRedirectCallback')
        .mockReturnValue(Promise.resolve(mockUser));

      const user = await authService.endLogin();

      expect(user).toBe(mockUser);
    });

    it('should call fetchApiToken with the user object', async () => {
      expect.assertions(1);
      jest.spyOn(authService, 'fetchApiToken');
      jest
        .spyOn(userManager, 'signinRedirectCallback')
        .mockResolvedValue(mockUser);

      await authService.endLogin();

      expect(authService.fetchApiToken).toHaveBeenNthCalledWith(1, mockUser);
    });

    it('should set the tokens and project ID in localStorage before the function returns', async () => {
      expect.assertions(4);
      jest
        .spyOn(userManager, 'signinRedirectCallback')
        .mockResolvedValue(mockUser);
      jest.spyOn(authService, 'queryApiTokensEndpoint').mockResolvedValue({
        data: {
          access_token,
          refresh_token,
        },
      });

      await authService.endLogin();

      expect(localStorage.setItem).toHaveBeenCalledTimes(3);
      expect(localStorage.setItem).toHaveBeenNthCalledWith(
        1,
        API_TOKEN,
        access_token
      );
      expect(localStorage.setItem).toHaveBeenNthCalledWith(
        2,
        REFRESH_TOKEN,
        refresh_token
      );
      expect(localStorage.setItem).toHaveBeenNthCalledWith(
        3,
        'projectId',
        testProjectId
      );
    });
  });

  describe('renewToken', () => {
    it('should call signinSilent from oidc', async () => {
      const signinSilent = jest
        .spyOn(userManager, 'signinSilent')
        .mockResolvedValue();

      await authService.renewToken();

      expect(signinSilent).toHaveBeenCalledTimes(1);
    });

    it('should resolve to the user value which has been resolved from signinSilent', async () => {
      expect.assertions(1);
      const mockUser = { name: 'Camilla Howe' };

      jest.spyOn(userManager, 'signinSilent').mockResolvedValueOnce(mockUser);

      const user = await authService.renewToken();

      expect(user).toBe(mockUser);
    });
  });

  describe('logout', () => {
    it('should call signoutRedirect from oidc', () => {
      const signoutRedirect = jest.spyOn(userManager, 'signoutRedirect');

      authService.logout();

      expect(signoutRedirect).toHaveBeenCalledTimes(1);
    });

    it('should remove the tokens from localStorage', async () => {
      expect.assertions(2);
      jest.spyOn(userManager, 'signoutRedirect').mockResolvedValue(undefined);
      const apiToken = 'a8d56df4-7ae8-4fbf-bf73-f366cd6fc479';
      const refreshToken = '347ed60c-88e4-4c08-ab25-9807be7666f4';

      localStorage.setItem(API_TOKEN, apiToken);
      localStorage.setItem(REFRESH_TOKEN, refreshToken);

      await authService.logout();

      expect(localStorage.getItem(API_TOKEN)).toBeNull();
      expect(localStorage.getItem(REFRESH_TOKEN)).toBeNull();
    });

    it('should call clearStaleState', async () => {
      expect.assertions(1);
      jest.spyOn(userManager, 'signoutRedirect').mockResolvedValue(undefined);
      jest.spyOn(userManager, 'clearStaleState').mockResolvedValue();

      await authService.logout();

      expect(userManager.clearStaleState).toHaveBeenCalledTimes(1);
    });

    it("should call authorization service's clear method", async () => {
      expect.assertions(1);
      jest.spyOn(userManager, 'signoutRedirect').mockResolvedValue(undefined);
      const authorizationServiceClearSpy = jest.spyOn(
        authorizationService,
        'clear'
      );

      await authService.logout();

      expect(authorizationServiceClearSpy).toHaveBeenCalledTimes(1);
    });
  });

  describe('fetchApiToken', () => {
    const access_token = 'db237bc3-e197-43de-8c86-3feea4c5f886';
    const mockUser = {
      name: 'Penelope Krajcik',
      access_token,
    };

    beforeEach(() => {
      axios.mockReset();

      axios.mockResolvedValue({
        data: {
          firstToken: '71ffd52c-5985-46d3-b445-490554f4012a',
          secondToken: 'de7c2a83-07f2-46bf-8417-8f648adbc7be',
        },
      });
    });

    it('should call axios with the right arguments', async () => {
      expect.assertions(2);
      await authService.fetchApiToken(mockUser);

      expect(axios).toHaveBeenCalledTimes(1);
      expect(axios.mock.calls[0]).toMatchSnapshot();
    });

    it('should call localStorage.setItem with the right arguments', async () => {
      expect.assertions(2);
      await authService.fetchApiToken(mockUser);

      expect(localStorage.setItem).toHaveBeenCalledTimes(1);
      expect(localStorage.setItem.mock.calls[0]).toMatchSnapshot();
    });
  });
});
