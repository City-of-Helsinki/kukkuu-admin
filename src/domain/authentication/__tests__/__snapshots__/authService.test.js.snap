// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authService fetchApiToken should call axios with the right arguments 1`] = `
Array [
  "https://tunnistus.test.hel.ninja/auth/realms/helsinki-tunnistus/protocol/openid-connect/token",
  Object {
    "baseURL": "https://tunnistus.test.hel.ninja/auth/realms/helsinki-tunnistus/",
    "data": Object {
      "audience": "kukkuu-api-dev",
      "grant_type": "urn:ietf:params:oauth:grant-type:uma-ticket",
      "permission": "#access",
    },
    "headers": Object {
      "Accept": "application/json",
      "Authorization": "bearer db237bc3-e197-43de-8c86-3feea4c5f886",
      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
    },
    "method": "post",
  },
]
`;

exports[`authService fetchApiToken should call localStorage.setItem with the right arguments 1`] = `
Array [
  "apiToken",
  undefined,
]
`;
